src/main/java/io/jawg/osmcontributor/flickr/oauth/OAuthRequest.java

```diff
@@ -58,7 +58,7 @@ public OAuthRequest(String apiKey, String apiKeySecret) {
     /*----------------CODE---------------------*/
     /*=========================================*/
     public void initParam(Map<String, String> params) {
-        this.params.clear();
+        this.params = new TreeMap<>();
```

Comment:
In fact, we need a TreeMap here, because OAuth parameters must be alphabetically sorted when creating signature. When looping over parameters, we must be sure that they are given in the right order, which is not guaranteed by HashMap, but is by TreeMap (which implemented SortedMap interface).